qutebrowser/utils/utils.py

```diff
@@ -773,3 +773,31 @@ def libgl_workaround() -> None:
     libgl = ctypes.util.find_library("GL")
     if libgl is not None:  # pragma: no branch
         ctypes.CDLL(libgl, mode=ctypes.RTLD_GLOBAL)
+
+
+def parse_duration(duration: str) -> int:
+    """Parse duration in format XhYmZs into milliseconds duration."""
+    if duration.isdigit():
+        # For backward compatibility return milliseconds
+        return int(duration)
+
+    match = re.search(
```

Comment:
If you use

```suggestion
    match = re.fullmatch(
```

here, you can get rid of the `^` and `$` in the regex.